src/lib/operators/case_operator.cpp

```diff
@@ -0,0 +1,226 @@
+   * _case_expression.
+   */
+  auto column_definitions = input_table_left()->column_definitions();
+  for (const auto& case_expression : _case_expressions) {
+    column_definitions.emplace_back("case_expr", case_expression->result_data_type, true);
+  }
+
+  const auto output_table = std::make_shared<Table>(column_definitions, TableType::Data);
+
+  /**
+   * For every Chunk in the input Table, create a Chunk in the output Table that contains the same columns as the input
+   * Table with the results from the CaseExpressions each as an additional Column
+   */
+  for (ChunkID chunk_id{0}; chunk_id < input_table_left()->chunk_count(); ++chunk_id) {
+    const auto input_chunk = input_table_left()->get_chunk(chunk_id);
+
+    /**
+     * Case appends a new column for every CaseExpression. All input columns are forwarded, and as Case produces a
+     * materialized Data Table, we re-use the columns from the input chunk if they are Data Columns and materialize them
+     * if they are Reference Columns.
+     */
+    ChunkColumns output_columns;
+    if (input_table_left()->type() == TableType::References) {
+      for (const auto& column : input_chunk->columns()) {
+        output_columns.emplace_back(materialize_as_value_column(*column));
+      }
+    } else {
+      output_columns = input_chunk->columns();
+    }
+
+    /**
+     * Create an output_column from each CaseExpression
+     */
+
+    // If a column in the input chunk is used as a WHEN column, it is materialized into this cache
+    std::vector<std::optional<std::vector<int32_t>>> materialized_when_columns(input_table_left()->column_count());
+
+    for (const auto& abstract_case_expression : _case_expressions) {
+      resolve_data_type(abstract_case_expression->result_data_type, [&](auto type) {
+        using ResultDataType = typename decltype(type)::type;
+        using NullOrValue = std::pair<bool, ResultDataType>;
+
+        // If a column in the input chunk is used as a THEN column, it is materialized into this cache
+        std::vector<std::optional<std::vector<NullOrValue>>> materialized_then_columns(
+            input_table_left()->column_count());
+
+        const auto& case_expression =
+            static_cast<const PhysicalCaseExpression<ResultDataType>&>(*abstract_case_expression);
+
+        pmr_concurrent_vector<ResultDataType> output_values(input_chunk->size());
+        pmr_concurrent_vector<bool> output_nulls(input_chunk->size());
+
+        /**
+         * Compute the CaseExpression for each row in the Chunk.
+         */
+        for (auto chunk_offset = ChunkOffset{0}; chunk_offset < input_chunk->size(); ++chunk_offset) {
+          auto any_when_claused_fulfilled = false;  // If no WHEN clause was TRUE, we have to use the ELSE clause
+
+          /**
+           * Find the first case_clause whose WHEN is true and write its THEN to output_values/nulls
+           */
+          for (const auto& case_clause : case_expression.clauses) {
```

Comment:
According to line 65, `.clauses` should be `CaseWhenClause`. See what naming problem I’m hinting at?